src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/LocalizationModelDeserializer.cs

```diff
@@ -0,0 +1,230 @@
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.TemplateEngine.Orchestrator.RunnableProjects
+{
+    internal static class LocalizationModelDeserializer
+    {
+        private const string PostActionIndexPrefix = "postActions[";
+        private const string PostActionIndexSuffix = "]";
+
+        private const string ManualInstructionIndexPrefix = "manualInstructions[";
+        private const string ManualInstructionIndexSuffix = "]";
+
+        public static ILocalizationModel Deserialize(JObject data)
+        {
+            var locModel = new LocalizationModel();
+            var parameterLocalizations = new Dictionary<string, ParameterSymbolLocalizationModel>();
+
+            // Property names are in format: symbols.framework.choices.0.description
+            // Split them using '.' and store together with the localized string (property value).
+            IEnumerable<(IEnumerable<string> nameParts, string localizedString)> stringsWithNames = data.Properties()
+                .Where(p => p.Value.Type == JTokenType.String)
+                .Select(p => (p.Name.Split('.').AsEnumerable(), p.Value.ToString()))
+                .ToList();
+
+            var symbols = LoadSymbolModels(stringsWithNames
+                .Where(s => s.nameParts.FirstOrDefault() == "symbols")
+                .Select(s => (s.nameParts.Skip(1), s.localizedString)));
+
+            var postActions = LoadPostActionModels(stringsWithNames
+                .Where(s => s.nameParts.FirstOrDefault().StartsWith(PostActionIndexPrefix))
+                .Select(s => (s.nameParts, s.localizedString)));
+
+            return new LocalizationModel()
+            {
+                Author = stringsWithNames.FirstOrDefault(s => s.nameParts.SingleOrDefault() == "author").localizedString,
+                Name = stringsWithNames.FirstOrDefault(s => s.nameParts.SingleOrDefault() == "name").localizedString,
+                Description = stringsWithNames.FirstOrDefault(s => s.nameParts.SingleOrDefault() == "description").localizedString,
+                ParameterSymbols = symbols,
+                // TODO uncomment once type compatibility issues are fixed.
+                // PostActions = postActions,
+            };
+        }
+
+        /// <summary>
+        /// Generates parameter symbol localization models. The given name parts should begin with the parameter name
+        /// as shown below and should not contain the string "symbols".
+        /// <list type="table">
+        /// <item>framework.displayName</item>
+        /// <item>framework.description</item>
+        /// <item>framework.choices.net5_0.description</item>
+        /// <item>targetframeworkoverride.description</item>
+        /// </list>
+        /// </summary>
+        private static IReadOnlyDictionary<string, IParameterSymbolLocalizationModel> LoadSymbolModels(IEnumerable<(IEnumerable<string> nameParts, string localizedString)> strings)
```

Comment:
I feel like name parts can be identified in this method instead, and starts with symbols and we do extraction in this method; same for similar methods. 
The input parameter should be probably json or parsed string collection (as string, localized string) and the method should filter relevant ones.

Reply:
That was my first go to. However, I ended up doing the same thing in multiple places - creating duplicated code. Any property that goes deeper than the first level will need this.

Reply:
I think Vlada is refering to:
```
LoadPostActionModels(stringsWithNames
                .Where(s => s.nameParts.FirstOrDefault().StartsWith(PostActionIndexPrefix))
                .Select(s => (s.nameParts, s.localizedString)));
```
should become:
`LoadPostActionModels(stringsWithNames);`
but same for Symbols...

Reply:
Yes, I think I understand right. I'm just not sure how that is better. I don't see the encapsulation benefit we are getting from moving this code in between private methods within the same class. And this is also at the expense of code duplication.
Regardless, I updated the methods to move forward.